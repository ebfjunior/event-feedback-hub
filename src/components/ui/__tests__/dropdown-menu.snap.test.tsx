import React from 'react';
import { render } from '@testing-library/react';
import {
	DropdownMenu,
	DropdownMenuContent,
	DropdownMenuItem,
	DropdownMenuLabel,
	DropdownMenuTrigger,
} from '../dropdown-menu';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return {
		...actual,
		CheckIcon: (p: any) => <svg data-icon="check" {...p} />,
		ChevronRightIcon: (p: any) => <svg data-icon="chev-right" {...p} />,
		CircleIcon: (p: any) => <svg data-icon="circle" {...p} />,
	};
});

describe('ui/DropdownMenu (snapshot)', () => {
	it('renders structure', () => {
		render(
			<DropdownMenu>
				<DropdownMenuTrigger>Open</DropdownMenuTrigger>
				<DropdownMenuContent>
					<DropdownMenuLabel>Label</DropdownMenuLabel>
					<DropdownMenuItem>Item</DropdownMenuItem>
				</DropdownMenuContent>
			</DropdownMenu>,
		);
		expect(document.body).toMatchSnapshot();
	});
});


