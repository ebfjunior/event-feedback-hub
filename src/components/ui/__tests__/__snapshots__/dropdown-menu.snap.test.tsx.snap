// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/DropdownMenu (snapshot) > renders structure 1`] = `
<body>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      data-slot="dropdown-menu-trigger"
      data-state="closed"
      id="radix-Â«r0Â»"
      type="button"
    >
      Open
    </button>
  </div>
</body>
`;
